packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js

```diff
@@ -331,9 +289,13 @@ const ApiTokenCreateView = () => {
             lifespan: apiToken?.lifespan,
           }}
           enableReinitialize
-          onSubmit={(body, actions) => handleSubmit(body, actions, tokenTypeValue)}
+          onSubmit={(body, actions) => handleSubmit(body, actions)}
         >
           {({ errors, handleChange, isSubmitting, values }) => {
+            if (state.selectedActions.length > 0 && !values?.type) {
+              values.type = 'custom';
```

Comment:
Mutating a render prop inside the render seems weird to me. Is there a way to do it it in an event listener for example?

Reply:
No unfortunately because I need to overwrite the api token type to custom only when we create an api token and when we select for the first time a permission (I need to have access to the form state, values in the example)

Reply:
I can set an initial value, I think can be a better solution